src/mango/src/mango_cursor_text.erl

```diff
@@ -334,4 +350,805 @@ get_json_docs(DbName, Hits) ->
+        },
+    Hit1 = #sortable{item = #hit{fields = [{<<"_id">>, {id, 1}}]}},
+    Hit2 = #sortable{item = #hit{fields = [{<<"_id">>, {id, 2}}]}},
+    Hit3 = #sortable{item = #hit{fields = [{<<"_id">>, {id, 3}}]}},
+    Hits = [Hit1, Hit2, Hit3],
+    meck:expect(
+        dreyfus_fabric_search,
+        go,
+        [db_name, IdxDDoc, IdxName, QueryArgs],
+        meck:val({ok, [bookmark, 1], undefined, Hits, undefined, undefined})
+    ),
+    meck:expect(mango_selector_text, convert, [selector], meck:val(query)),
+    meck:expect(mango_selector, match, fun(selector, {doc, _N}) -> true end),
+    meck:expect(
+        foo,
+        stops,
+        fun(Args, {acc, N}) ->
+            case Args of
+                {row, {final_doc, _M}} -> ok;
+                {add_key, bookmark, [bookmark, _M]} -> ok
+            end,
+            Status =
+                case N of
+                    2 -> stop;
+                    _ -> ok
+                end,
+            {Status, {acc, N + 1}}
+        end
+    ),
+    ?assertEqual({ok, {acc, 6}}, execute(Cursor, fun foo:stops/2, {acc, 0})),
+    ?assertEqual(UniqueHits, meck:num_calls(couch_stats, increment_counter, 1)),
+    ?assertEqual(UniqueHits, meck:num_calls(mango_execution_stats, incr_docs_examined, 1)),
+    ?assertEqual(3, meck:num_calls(mango_execution_stats, incr_results_returned, 1)).
+
+t_execute_search_error(_) ->
+    Limit = 10,
+    Skip = 0,
+    IdxDDoc = <<"ddoc">>,
+    IdxName = <<"index">>,
+    Options = [{partition, <<>>}, {sort, {[]}}, {bookmark, [bookmark, 0]}],
+    Cursor = #cursor{
+        db = db,
+        index = #idx{ddoc = IdxDDoc, name = IdxName},
+        limit = Limit,
+        skip = Skip,
+        fields = fields,
+        selector = selector,
+        opts = Options,
+        execution_stats = stats
+    },
+    QueryArgs =
+        #index_query_args{
+            q = query,
+            partition = nil,
+            limit = Skip + Limit,
+            bookmark = [bookmark, 0],
+            sort = relevance,
+            raw_bookmark = true
+        },
+    meck:expect(
+        dreyfus_fabric_search,
+        go,
+        [db_name, IdxDDoc, IdxName, QueryArgs],
+        meck:val({error, reason})
+    ),
+    meck:expect(mango_selector_text, convert, [selector], meck:val(query)),
+    Exception = {mango_error, mango_cursor_text, {text_search_error, {error, reason}}},
+    ?assertThrow(Exception, execute(Cursor, fun foo:normal/2, {acc, 0})),
+    ?assertEqual(0, meck:num_calls(dreyfus_fabric, get_json_docs, 2)),
+    ?assertEqual(0, meck:num_calls(foo, normal, 2)),
+    ?assertEqual(0, meck:num_calls(couch_stats, increment_counter, 1)),
+    ?assertEqual(0, meck:num_calls(mango_execution_stats, incr_docs_examined, 1)),
+    ?assertEqual(0, meck:num_calls(mango_execution_stats, incr_results_returned, 1)).
+
+explain_test_() ->
+    {
+        foreach,
+        fun() ->
+            ok
+        end,
+        fun(_) ->
+            meck:unload()
+        end,
```

Comment:
Tiny functions like this in setup/tear-down can just be one-liners. Up to you though, if you prefer this style for consistency or erlfmt enforces it that's fine too.

Reply:
I think I will change it to a one-liner.